src/main/java/edu/harvard/iq/dataverse/harvest/client/ResetHarvestInProgressBean.java

```diff
@@ -0,0 +1,23 @@
+package edu.harvard.iq.dataverse.harvest.client;
+
+import javax.annotation.PostConstruct;
+import javax.ejb.EJB;
+import javax.ejb.Singleton;
+import javax.ejb.Startup;
+
+@Singleton
```

Comment:
Maybe I'm just being paranoid but we've had trouble with Singleton beans recently. See https://github.com/IQSS/dataverse.harvard.edu/issues/73

Reply:
I read the issue you linked very closely since the last thing I want to do is introduce new errors. In the issue there's an authentication bean that's a singleton and when the bean gets held up by a process, no other clients can use it to authenticate. This is bad. However, in my use case the singleton is only used during the startup of Dataverse to reset any hanging harvesting clients. After that there's no use for it, so a singleton bean seems the right approach here. Moreover, in Dataverse there's many singleton beans with an additional @Startup tag to initialize a certain process and those don't seem to be causing issues either. So I'm fairly confident that a singleton is the way to go here.

Reply:
I'm not worried about it being a singleton - seeing how it has nothing in it but the `@PostConstruct` code. 
But there's another reason why this is not going to work, specifically for us here at Harvard. We have 2 production servers. So if one of the two is being restarted, it'll reset the flags even though the actions in question may still be in progress on the other node. 

It's easy to address **for the most part**: in a multi-server situation only one node is designated to do harvesting, and other timer-run jobs. This "master node" is the one that has the JVM option `-Ddataverse.timerServer=true`. And you can use `if (systemConfig.isTimerServer())` to check if this node is the timer node in your code, before proceeding to clear the flags. 

HOWEVER, there is an exception to the rule - when a harvesting run is started not by the timer, but manually, through the harvesting client page and/or API, the job can end up running on either of the multiple nodes. 

Admittedly,  this will not affect many otheaverse installations. (I'm actually not aware of another production installation at this point that is using a multiple server setup). But this would affect us, so I am a bit hesitant about doing this. 
And I'm not really positive as to what  the best reliable solution is. That ClientHarvestRun object would need an extra entry identifying the node actually running the harvest, maybe?

Reply:
Another solution would be to change how the manually-run harvests are handled. I.e., instead of actually running the harvest it could create a one-time timer entry for 5 sec. from now...

Reply:
I'm not 100% sure if being able to have these harvest-related  flags cleared automatically is worth adding any complicated code to address the multiple server scenario... 
So yet another, much simpler, solution would be to implement it as above - but with an option to disable this mechanism; so on startup the service above would check the setting before proceeding - `if (enableClearingHarvestJobsOnStartup) { ... }`. It will be enabled by default; so it would work for most installations. But then in the guide it will say "make sure to disable this mechanism if you are running multiple servers!". Somewhere in the same part of the guide where we mention that `dataverse.timerServer`setting.

Reply:
@landreev My plan is to add a sane default for a configurable DB key "EnableClearingHarvestJobsOnStartup" using the SettingsServiceBean, do you have any objections?